### PR TITLE
DM-24844: Migrate ap_verify_testdata to obs_lsst

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR adds explicit ImSim support to the dataset conversion script, and fixes a config bug that affects instruments that exclude files from conversion by default.

### CODE REVIEW

**Comment by kfindeisen** [comment]
I've also added a fix in response to [DM-24804](https://jira.lsstcorp.org/browse/DM-24804).

File: scripts/add_gen3_repo.py [hunk 1]
```diff
@@ -96,14 +96,14 @@ def _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calib_repo, gen3_repo, mode,
...
-    # Call convertGepoToGen3 instead of calling ConvertRepoTask directly, to
+    # Call the script instead of calling ConvertRepoTask directly, to
     # avoid manually having to do a lot of setup that may change in the future
     # calib/<instrument>, refcats, and skymaps collections created by default
-    convertGen2RepoToGen3.convert(gen2_repo, gen3_repo, instrument,
-                                  # skymap detected automatically; do nothing
-                                  calibs=gen2_calib_repo,
-                                  config=config,
-                                  transferMode=mode)
+    convert(gen3_repo, gen2_repo, instrument,
```
Comment (parejkoj) [inline scripts/add_gen3_repo.py]: I would rather this be `lsst.obs.base.script.convert`, since `convert` is a pretty generic word.

**Review by parejkoj — COMMENTED** [review]
Was this template used to produce the `ap_verify_testdata` repo? If not, do they need to be kept in sync somehow?

**Comment by kfindeisen** [comment]
The template was (and will be?) used to create all the datasets. Syncing is a problem in general, but the `testdata` version needs some special-casing because it's a dataset that doesn't officially exist in `ap_verify`.

**Pull request merged**